test/prefer-global-this.mjs

```diff
@@ -0,0 +1,173 @@
+import {getTester} from './utils/test.mjs';
+import outdent from 'outdent';
+
+const {test} = getTester(import.meta);
+
+test.snapshot({
+	valid: [
+		'globalThis',
+		'globalThis.foo',
+		'globalThis[foo]',
+		'globalThis.foo()',
+		'const { foo } = globalThis',
+		'function foo (window) {}',
+		'function foo (global) {}',
+		'var foo = function foo (window) {}',
+		'var foo = function foo (global) {}',
+		'var window = {}',
+		'let global = {}',
+		'const global = {}',
+		outdent`
+			function foo (window) {
+				window.foo();
+			}
+		`,
+		outdent`
+			var window = {};
+			function foo () {
+				window.foo();
+			}
+		`,
+		'foo.window',
+		'foo.global',
+		'import window from "xxx"',
+		'import * as window from "xxx"',
+		'import window, {foo} from "xxx"',
+		'export { window }  from "xxx"',
+		'export * as window from "xxx";',
+		outdent`
+			try {
+
+			} catch (window) {}
+		`,
+
+		// Use window specific APIs
+		'window.name = "foo"',
+		'window.addEventListener',
+		'window.innerWidth',
+		'window.innerHeight',
+		'self.location',
+		'self.navigator',
+		'window.addEventListener("resize", () => {})',
+		'window.onresize = function () {}',
+		outdent`
+			const {window} = jsdom()
+			window.jQuery = jQuery;
+		`,
+		'({ foo: window.name } =  {})',
+		'[window.name] = []',
+	],
+	invalid: [
+		'global',
+		'self',
+		'window',
+		'window.foo',
+		'window[foo]',
+		'window[title]',
+		'window["foo"]',
```

Comment:
These should be valid, since they are computed.

Reply:
Update in e3e48c070298b61520b659c195c69ce1e86bb8a3